### ISSUE TITLE
UI not refreshing to login screen after session token is expiered

### ISSUE THREAD

**Issue by tal-hason** [akuity/kargo#877]
when the UI is left open for a while the session token expires and the UI still shows the last view without updates.

if the token expires the UI should return to the login screen without the user needing to refresh twice( 1st time to get the token expire error, 2nd to return to the login screen)

also, add a session token timeout value in the helm chart(API server)

**Comment by krancour** [comment]
Can I ask for some clarification? We're sessionless. Are you talking about the _identity_ token?

**Comment by jessesuen** [comment]
Yes, I'm sure he means the token used for auth. I've actually seen it work as expected, where I get redirected back to login after coming back the next day, but will confirm the next time I try this.

I suspect that UI needs to perform some API interaction before detecting its token is bad. But perhaps it could do that by just decoding the jwt and understanding its expiry before an failed API call.

**Comment by krancour** [comment]
I've seen it work correctly as well, but have sometimes seen it be a bit sluggish to catch on.

I like the idea of decoding the JWT.

**Comment by krancour** [comment]
> also, add a session token timeout value in the helm chart(API server)

Obviously, there's an issue to address here, but just fyi, admin token TTL can already be configured in the chart.

Its default is already increased to 24h as of #845 

For SSO, it's an identity provider setting if it's configurable at all.

**Comment by krancour** [comment]
Adding to the idea of checking token expiry... that would also be an appropriate place to automatically use the refresh if there is one (and if we aren't already).

@rbreeze or @rpelczar does one of you want to take a look at this issue?

**Comment by krancour** [comment]
@rpelczar assigning to you just so someone's on point. Feel free to re-assign or re-prioritize.

**Comment by smolinari** [comment]
I've also come up to this issue, but also the issue that the invalid token warning comes up after about 5 minutes. I haven't tested it to know the time exactly, but that is the around about feeling, as it is pretty short term, so fairly annoying. It's like there is no refresh process. The access token expires and I'm forced to refresh via login via clicking the SSO button. 

If this should be its own issue, let me know, I'll start one. 

Scott
